initialize_package.py

```diff
@@ -0,0 +1,325 @@
+#!/usr/bin/env python3
+import os
+import re
+import subprocess
+from pathlib import Path
+
+from initialization_wizard_utils.filesystem import (
+    rm_files,
+    sed_in_place,
+    sed_remove_line_matching,
+    sed_remove_range,
+    replace_text_in_file,
+    traverse_and_rename,
+)
+from initialization_wizard_utils.globals import (
+    TEMPLATE_SOURCES,
+    OG_CONTROLLER_PACKAGE,
+    OG_CONTROLLER_NAME,
+    OG_COMPONENT_PACKAGE,
+    OG_COLLECTION_NAME,
+    PYTHON_COMPONENT_FILES,
+    CPP_COMPONENT_FILES,
+)
+
+PIP_REQUIREMENTS = {
+    "questionary": "==2.1.0",
+}
+
+
+def is_inside_docker() -> bool:
+    return os.environ.get("IN_DOCKER") == "1"
+
+
+def outside_docker_behavior():
+    script_path = Path(__file__).resolve()
+    script_dir = script_path.parent
+    docker_cmd = [
+        "docker",
+        "run",
+        "--rm",
+        "-it",
+        "-e",
+        "IN_DOCKER=1",
+        "-v",
+        f"{script_dir}:/{TEMPLATE_SOURCES}",
+        "python:3.12-slim",
+        "python",
+        f"/{TEMPLATE_SOURCES}/{script_path.name}",
+    ]
+    subprocess.run(docker_cmd, check=True)
+
+
+def inside_docker_behavior():
+    install_requirements()
+    run_wizard()
+
+
+def install_requirements():
+    path = "/tmp/requirements.txt"
+    with open(path, "w", encoding="utf-8") as f:
+        for pkg, version in PIP_REQUIREMENTS.items():
```

Comment:
* Given that you install within the same runtime that you want to use `questionnary` in, will Python be able to pick it up or do you need a restart?
* In general, I don't think it's a good idea to automatically install libraries on the user's global pip setup (which is what this does unless they spin up a virtualenv before calling the script). This is actually forbidden on macOS now (and some Linux distros I have used as well) as it might mess up the system Python setup. Either we need to keep a shellscript as a wrapper which sets up the virtualenv or reimplement `questionnary` using the stdlib.

Reply:
Hm, not sure if I am understanding this correctly. 

Is you suggestion to keep the shell script that sets up a docker container with an entrypoint that starts the Python script? I guess you'd like to remove the docker logic from the Python script?